src/session_handle.cpp

```diff
@@ -201,121 +201,128 @@ namespace libtorrent {
 #ifndef TORRENT_NO_DEPRECATE
 namespace {
 
-		void handle_backwards_compatible_resume_data(add_torrent_params& atp)
+	void handle_backwards_compatible_resume_data(add_torrent_params& atp)
+	{
+		// if there's no resume data set, there's nothing to do. It's either
+		// using the previous API without resume data, or the resume data has
+		// already been parsed out into the add_torrent_params struct.
+		if (atp.resume_data.empty()) return;
+
+		error_code ec;
+		add_torrent_params resume_data
+			= read_resume_data(atp.resume_data, ec);
+
+		resume_data.internal_resume_data_error = ec;
+		if (ec) return;
+
+		// now, merge resume_data into atp according to the merge flags
+		if (atp.flags & add_torrent_params::flag_use_resume_save_path
+			&& !resume_data.save_path.empty())
 		{
-			// if there's no resume data set, there's nothing to do. It's either
-			// using the previous API without resume data, or the resume data has
-			// already been parsed out into the add_torrent_params struct.
-			if (atp.resume_data.empty()) return;
-
-			error_code ec;
-			add_torrent_params resume_data
-				= read_resume_data(atp.resume_data, ec);
-
-			resume_data.internal_resume_data_error = ec;
-			if (ec) return;
-
-			// now, merge resume_data into atp according to the merge flags
-			if (atp.flags & add_torrent_params::flag_use_resume_save_path
-				&& !resume_data.save_path.empty())
-			{
-				atp.save_path = resume_data.save_path;
-			}
+			atp.save_path = std::move(resume_data.save_path);
+		}
 
-			if (!resume_data.trackers.empty())
-			{
-				atp.tracker_tiers.resize(atp.trackers.size(), 0);
-				atp.trackers.insert(atp.trackers.end()
-					, resume_data.trackers.begin()
-					, resume_data.trackers.end());
-				atp.tracker_tiers.insert(atp.tracker_tiers.end()
-					, resume_data.tracker_tiers.begin()
-					, resume_data.tracker_tiers.end());
-				if ((resume_data.flags & add_torrent_params::flag_merge_resume_trackers) == 0)
-					atp.flags |= add_torrent_params::flag_override_trackers;
-			}
+		if (!atp.ti)
```

Comment:
this is the only material change (the rest of this hunk is just reducing the indentation)